modules/auxiliary/scanner/smb/smb_version.rb

```diff
@@ -47,15 +47,10 @@ def initialize
       ]
     )
 
-    deregister_options('RPORT', 'SMBDIRECT', 'SMB::ProtocolVersion')
```

Comment:
I think we want to keep de-registering `SMB::ProtocolVersion`. The module makes some assumptions about how it tries to negotiate versions 1 2 and 3 in sequence to pull different pieces of information from each. IIRC, the server negotiates the highest dialect it supports so we have to ask for each to know if the server allows them.

Reply:
Good point, thank you. I added that `deregister_options` entry back in, using another commit. 

I don't use GitHub heavily enough to know that I did it right, though.